tests/cpu/test_kalman_fitter_wire_chamber.cpp

```diff
@@ -124,6 +124,7 @@ TEST_P(KalmanFittingWireChamberTests, Run) {
         static_cast<float>(mask_tolerance);
     fit_cfg.propagation.navigation.search_window = search_window;
     fit_cfg.ptc_hypothesis = ptc;
+    fit_cfg.covariance_inflation_factor = 1.f;
```

Comment:
Is this something specific for the wire chamber?

Reply:
It is for not changing the algorithm used in wire chamber test ( RTS smoother and no covariance inflation like in the past )

And yes, it is not working for the wire chamber after inflating covariance which will introduce an error in navigation result. Eventually, the surface in the track candidate won't be reached because cell of wire chamber is so small (~ 1 cm scale)

The navigator used for **forward** fitter needs a fix to make it find the sensitive surface correctly. (To be clear, I am not talking about the direct navigator). We need to make the mask tolerance for sensitive surfaces in the `track_candidate` be large enough.

Reply:
OK, interesting, thanks for explaining! I always found making the propagation integration test work for the wirechamber quite fiddly. Maybe we can start with the propagation conf used there. I am also worried by various overlaps that are still present in that geometry

Reply:
For the moment, making the mask tolerance bigger only for the track candidate surface seems to be enough to me. Making it bigger for all surfaces will be a disaster. I don't know if such selective tolerance setup is possible in detray

Reply:
From what I understand atm, we need a new mechanism in the navigator anyway that forces it to stop at some externally provided surfaces. Among them also the perigee perhaps